recipes/mysql-connector-cpp/all/conanfile.py

```diff
@@ -0,0 +1,167 @@
+        deps.set_property("protobuf::libprotobuf", "cmake_target_name", "ext::protobuf")
+        deps.set_property("protobuf::libprotobuf-lite", "cmake_target_name", "ext::protobuf-lite")
+        deps.set_property("rapidjson", "cmake_target_name", "RapidJSON::rapidjson")
+        deps.set_property("zlib", "cmake_target_name", "ext::z")
+        deps.set_property("lz4", "cmake_target_name", "ext::lz4")
+        deps.set_property("zstd", "cmake_target_name", "ext::zstd")
+        deps.generate()
+
+    def _patch_sources(self):
+        apply_conandata_patches(self)
+        # Disable boostrap(), which is unnecessary and fragile with variables set by Conan
+        # https://github.com/mysql/mysql-connector-cpp/blob/9.0.0/CMakeLists.txt#L69-L71
+        # https://github.com/mysql/mysql-connector-cpp/blob/9.0.0/cdk/cmake/bootstrap.cmake#L55
+        replace_in_file(self, os.path.join(self.source_folder, "CMakeLists.txt"), "bootstrap()", "")
+
+    def build(self):
+        self._patch_sources()
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def package(self):
+        copy(self, "LICENSE.txt", self.source_folder, os.path.join(self.package_folder, "licenses"))
+        cmake = CMake(self)
+        cmake.install()
+        rm(self, "INFO_SRC", self.package_folder)
+        rm(self, "INFO_BIN", self.package_folder)
+        rm(self, "*.cmake", self.package_folder)
+
+    def package_info(self):
+        self.cpp_info.set_property("cmake_file_name", "mysql-concpp")
+        self.cpp_info.set_property("cmake_target_name", "mysql::concpp")
+
+        aliases = ["mysql::concpp-xdevapi"]
+        if not self.options.shared:
+            aliases.append("mysql::concpp-static")
+            aliases.append("mysql::concpp-xdevapi-static")
+            if self.settings.build_type == "Debug":
+                aliases.append("mysql::concpp-static-debug")
+                aliases.append("mysql::concpp-xdevapi-static-debug")
+        aliases.append("mysql::openssl")
+        self.cpp_info.set_property("cmake_target_aliases", aliases)
+
+        lib = "mysqlcppconnx"
+        if not self.options.shared:
+            lib += "-static"
+            if is_msvc_static_runtime(self):
+                lib += "-mt"
+        self.cpp_info.libs = [lib]
+
+        if self.settings.os == "Windows":
+            self.cpp_info.system_libs.extend(["dnsapi", "ws2_32"])
+        elif self.settings.os != "FreeBSD":
+            self.cpp_info.system_libs.append("resolv")
+        if self.settings.os == "SunOS":
+            self.cpp_info.system_libs.append(["socket", "nsl"])
+
+        if not self.options.shared:
+            self.cpp_info.defines = ["MYSQL_STATIC", "STATIC_CONCPP"]
+
+
+        if is_apple_os(self) or self.settings.os in ["Linux", "FreeBSD"]:
+            self.cpp_info.system_libs.extend(["resolv"])
+            if self.settings.os in ["Linux", "FreeBSD"]:
+                self.cpp_info.system_libs.extend(["m", "pthread"])
```

Comment:
```suggestion
                self.cpp_info.system_libs.extend(["m", "pthread", "dl"])
```

https://github.com/mysql/mysql-connector-cpp/blob/9.0.0/mysql-concpp-config.cmake.in#L463